### DISCUSSION TITLE
Gamepad support for osu!lazer on Android

### DISCUSSION THREAD

**Discussion by CoolGuy12490** [ppy/osu#18856] [Ideas] [answered]
If gamepad support was added, it would be a **BIG** help to anyone not willing to get a Bluetooth keyboard or a USB keyboard with an adapter.

**Comment by frenzibyte** [comment]
Native gamepad support is not available for mobile platforms due to still relying on a different backend (osuTK/OpenTK) than what desktop platforms use (SDL). We'll eventually get to update them to use SDL.

However, I'm not sure if Android allows simulating gamepad buttons to keyboard (likely not).

**Reply by bdach** [reply]
Controllers can be handled on android without SDL2 via pathways we already use for mouse. See: https://developer.android.com/training/game-controllers/controller-input

So the above is wrong, there is work that can be done here.

**Reply by frenzibyte** [reply]
Seems like an alternative option that would work for the time being. Not sure that necessarily means the above is "wrong".

**Comment by smoogipoo** [comment, marked as answer]
Will be resolved via https://github.com/ppy/osu-framework/pull/5281
